ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -61,11 +76,39 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
   );
 
   const onSaveButtonClick = () => {
+    const { newVariables, isSelectedVariablesUpdated } = updateVariableDefaultValues(savedVariables, variableValues);
```

Comment:
Do you think it makes sense to make a `SaveButton` component, similar to how we export `EditJsonButton` and `AddPanelButton`? There is a lot of logic related to this save button now, so I am thinking it might be better if we encapsulate this in its own file